File: WebAPI/Controllers/BeaconController.cs
using System.Threading.Tasks;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using ReachUp;

namespace ReachUpWebAPI.Controllers
{
    [EnableCors("CorsPolicy")]
    [Route("api/[controller]")]
    [ApiController]
    public class BeaconController : ControllerBase
    {

        [Authorize(Roles = "adm, dev")]
        [HttpGet("GetAll")]
        public async Task<IActionResult> GetAll(int type)
        {
            if (!string.IsNullOrWhiteSpace(type.ToString()))
                return Ok(await new Beacon().GetAll(type));
            return BadRequest("Parameters are null");
        }

        [Authorize(Roles = "adm,dev")]
        [HttpGet]
        public async Task<IActionResult> Get(string uuid)
        {
            if (!string.IsNullOrWhiteSpace(uuid))
                return Ok(await new Beacon().Get(uuid));
            return BadRequest("Parameters are null");
        }

        [Authorize(Roles = "adm, dev")]
        [HttpGet("ByLocal")]
        public async Task<IActionResult> ByLocal(int local)
        {
            if (!string.IsNullOrWhiteSpace(local.ToString()))
                return Ok(await new Beacon().ByLocal(local));
            return BadRequest("Parameters are null");
        }


        [Authorize(Roles = "adm,dev")]
        [HttpPost]
        public async Task<IActionResult> Post([FromBody] Beacon beacon)
        {
            if (beacon != null)
                return Ok(await beacon.Add());
            return BadRequest("Parameters are null");
        }

        [Authorize(Roles = "adm, dev")]
        [HttpPatch]
        public async Task<IActionResult> Patch([FromBody] Beacon beacon)
        {
            if (beacon != null)
                return Ok(await beacon.Update());
            return BadRequest("Parameters are null");
        }

        [Authorize(Roles = "adm, dev")]
        [HttpDelete]
        public async Task<IActionResult> Delete(string uuid)
        {
            if (!string.IsNullOrWhiteSpace(uuid))
                return Ok(await new Beacon().Delete(uuid));
            return BadRequest("Parameters are null");
        }
    }
}

File: WebDev/src/Controllers/BeaconController.ts
import { Beacon } from "../Models/Beacon";
import { BeaconType } from "../Models/BeaconType";
import { Local } from "../Models/Local";
import { ApiConnection } from "../services/ApiConnection";

export module BeaconController {

  export class clsBeaconController extends ApiConnection.clsApiConnection {
        
    constructor(){
       super();
    }

    public async Get(uuid : string)
    {
       return await this.httpGet(`Beacon/Get?uuid=${uuid}`);
    }

    public async GetAll(type : number)
    {
       return await this.httpGet(`Beacon/GetAll?type=${type}`);
    }

    public async ByLocal(id : number)
    {
       return await this.httpGet(`Beacon/ByLocal?local=${id}`);
    }

    public async Post(uuid : string, type : BeaconType.clsBeaconType, 
      local : Local.clsLocal)
    {
       return await this.httpPost('Beacon/Post',
         new Beacon.clsBeacon(uuid, type, local)
        );
    }

    public async Patch(uuid : string, type : BeaconType.clsBeaconType, 
      local : Local.clsLocal)
    {
       return await this.httpPatch('Beacon/Patch',
         new Beacon.clsBeacon(uuid, type, local));
    }

    public async Delete(uuid : string)
    {
       return await this.httpDelete(`Beacon/Delete?uuid=${uuid}`);
    }
  }
}
